app/devUtilities/validate-areas.js

```diff
@@ -0,0 +1,229 @@
+/* eslint-disable import/no-extraneous-dependencies, no-console, no-restricted-syntax */
+const { Command } = require('commander');
+const mongoose = require('mongoose');
+const fs = require('fs');
+const cliProgress = require('cli-progress');
+const Area = require('models/area.modelV2');
+
+// Setup Commander for CLI options
+const program = new Command();
+
+program
+    .name('validate-areas')
+    .description('CLI tool to validate and optionally save all documents in the Area collection.')
+    .option('-h, --host <host>', 'MongoDB host', 'localhost')
+    .option('-p, --port <port>', 'MongoDB port', '27017')
+    .option('-d, --database <database>', 'MongoDB database name', 'mydatabase')
+    .option('-u, --username <username>', 'MongoDB username', 'admin')
+    .option('-P, --password <password>', 'MongoDB password', 'password123')
+    .option('-b, --batch-size <number>', 'Number of documents to process in each batch', 500)
+    .option('--dryrun', 'Skip saving changes and only run validation')
+    .option('--logfile <path>', 'Save logs to a specified file')
+    .parse(process.argv);
+
+// Get options from the CLI
+const options = program.opts();
+
+// Build the MongoDB URI
+const mongoURI = `mongodb://${options.username}:${options.password}@${options.host}:${options.port}/${options.database}?directConnection=true&retryWrites=false`;
+
+// Batch size
+const BATCH_SIZE = parseInt(options.batchSize, 10);
+
+// Create a write stream for logging if --logfile is provided
+let logStream = null;
+if (options.logfile) {
+    logStream = fs.createWriteStream(options.logfile, { flags: 'a' });
+    console.log(`📄 Logs will be saved to: ${options.logfile}`);
+}
+
+// Function to log messages to both console and file
+function logMessage(message) {
+    if (!options.logfile) {
+        console.log(message);
+    }
+    if (logStream) {
+        logStream.write(`${message}\n`);
+    }
+}
+
+// Function to always log metrics to both console and file
+function logStats(message) {
+    console.log(message);
+    if (logStream) {
+        logStream.write(`${message}\n`);
+    }
+}
+
+// Statistics
+let totalDocuments = 0;
+let totalValidated = 0;
+let totalFailures = 0;
+let totalAdminBoundaries = 0;
+
+// Function to check if a document is an Administrative Boundary
+function isAdministrativeBoundary(area) {
+    return area.admin?.adm0 || area.iso?.country;
+}
+
+// Function to connect to MongoDB
+async function connectDB() {
+    const configs = {
+        useNewUrlParser: true,
+        useUnifiedTopology: true,
+        serverSelectionTimeoutMS: 10000,
+        socketTimeoutMS: 60000,
+        keepAlive: true,
+        keepAliveInitialDelay: 300000,
+    };
+
+    try {
+        await mongoose.connect(mongoURI, configs);
+
+        mongoose.connection.on('disconnected', () => {
+            logStats('🔌  MongoDB connection lost. Attempting to reconnect...');
```

Comment:
Love the emojis, how exactly are making them work in a console application?

Reply:
I think it makes coding way more fun! 😄 

Here is a gist of what's supported in bash and zsh: https://gist.github.com/BuonOmo/77b75349c517defb01ef1097e72227af